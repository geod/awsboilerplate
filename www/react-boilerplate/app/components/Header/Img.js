import styled from 'styled-components';

import NormalImg from 'components/Img';

const Img = styled(NormalImg)`
  width: 100%;
  margin: 0 auto;
  display: block;
`;

export default Img;
