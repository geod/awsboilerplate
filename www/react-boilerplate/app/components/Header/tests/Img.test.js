import React from 'react';
import { render } from 'react-testing-library';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Img from '../Img';

describe('<Img />', () => {
  it('should match the snapshot', () => {
    const renderedComponent = renderer
      .create(<Img src="http://example.com/test.jpg" alt="test" />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should have a class attribute', () => {
    const { container } = render(
      <Img src="http://example.com/test.jpg" alt="test" />,
    );
    expect(container.querySelector('img').hasAttribute('class')).toBe(true);
  });

  it('should adopt a valid attribute', () => {
    const { container } = render(
      <Img src="http://example.com/test.jpg" alt="test" />,
    );
    expect(container.querySelector('img').alt).toEqual('test');
  });

  it('should not adopt an invalid attribute', () => {
    const { container } = render(
      <Img src="http://example.com/test.jpg" attribute="test" alt="test" />,
    );
    expect(container.querySelector('img').getAttribute('attribute')).toBeNull();
  });
});
