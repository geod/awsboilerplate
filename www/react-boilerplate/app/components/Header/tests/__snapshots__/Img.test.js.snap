// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Img /> should match the snapshot 1`] = `
.c0 {
  width: 100%;
  margin: 0 auto;
  display: block;
}

<img
  alt="test"
  className="c0"
  src="http://example.com/test.jpg"
/>
`;
